core/src/mindustry/entities/def/TileComp.java

```diff
@@ -55,6 +55,7 @@
     PowerModule power;
     ItemModule items;
     LiquidModule liquids;
+    ThroughputModule throughput;
```

Comment:
This is a very specific use case; it shouldn't be a module. Better to have it in each relevant void class instead.

Reply:
that was my first thought, but having 4 identical bar codes scattered around as duplicate seemed bad imo?

Reply:
The throughput class could still exist with some utility methods (like adding bars), but without extending `BlockModule` and only in the void entities.

Reply:
where would you like the throughput class to be? an interface like autotiler?

Reply:
Keep it the way it is, but remove the `BlockModule` superclass, add some utility methods and add fields to the void entities containing an instance of it.

Reply:
how would the utility methods know that the entities have those two variables?

Reply:
and how will the addbars on the block access the module inside the entity to add them?